crypto/storage/vault/vault.go

```diff
@@ -60,6 +61,7 @@ func DefaultConfig() Config {
 // logicaler is an interface which has been implemented by the mockVaultClient and real vault.Logical to allow testing vault without the server.
 type logicaler interface {
 	Read(path string) (*vault.Secret, error)
+	ReadWithDataWithContext(ctx context.Context, path string, data map[string][]string) (*vault.Secret, error)
```

Comment:
when should `ReadWithDataWithContext` be used over `Read`? 
`data` seems to be unused, can we use `ReadWithContext` instead?